src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -1083,42 +1083,67 @@ void LottieBuilder::updateText(LottieLayer* layer, float frameNo)
 
                         shape->opacity((*s)->style.opacity(frameNo));
 
-                        auto color = (*s)->style.fillColor(frameNo);
-                        shape->fill(color.rgb[0], color.rgb[1], color.rgb[2], (*s)->style.fillOpacity(frameNo));
+                        auto rangeColor = (*s)->style.fillColor(frameNo);
+                        if (tvg::equal(f, 1.0f)) shape->fill(rangeColor.rgb[0], rangeColor.rgb[1], rangeColor.rgb[2], (*s)->style.fillOpacity(frameNo));
+                        else {
+                            auto color = doc.color;
+                            if (setColor) {
+                                color.rgb[0] = P(shape)->rs.color[0];
+                                color.rgb[1] = P(shape)->rs.color[1];
+                                color.rgb[2] = P(shape)->rs.color[2];
+                            }
+                            auto r = lerp<uint8_t>(color.rgb[0], rangeColor.rgb[0], f);
+                            auto g = lerp<uint8_t>(color.rgb[1], rangeColor.rgb[1], f);
+                            auto b = lerp<uint8_t>(color.rgb[2], rangeColor.rgb[2], f);
+                            shape->fill(r, g, b, (*s)->style.fillOpacity(frameNo));
+                        }
 
                         if (doc.stroke.render) {
-                            auto strokeColor = (*s)->style.strokeColor(frameNo);
-                            shape->strokeWidth((*s)->style.strokeWidth(frameNo) / scale);
-                            shape->strokeFill(strokeColor.rgb[0], strokeColor.rgb[1], strokeColor.rgb[2], (*s)->style.strokeOpacity(frameNo));
+                            shape->strokeWidth(f * (*s)->style.strokeWidth(frameNo) / scale);
+                            auto rangeColor = (*s)->style.strokeColor(frameNo);
+                            if (tvg::equal(f, 1.0f)) shape->strokeFill(rangeColor.rgb[0], rangeColor.rgb[1], rangeColor.rgb[2], (*s)->style.strokeOpacity(frameNo));
+                            else {
+                                auto color = doc.stroke.color;
+                                if (setColor) {
+                                    color.rgb[0] = P(shape)->rs.stroke->color[0];
+                                    color.rgb[1] = P(shape)->rs.stroke->color[1];
+                                    color.rgb[2] = P(shape)->rs.stroke->color[2];
+                                }
+                                auto r = lerp<uint8_t>(color.rgb[0], rangeColor.rgb[0], f);
+                                auto g = lerp<uint8_t>(color.rgb[1], rangeColor.rgb[1], f);
+                                auto b = lerp<uint8_t>(color.rgb[2], rangeColor.rgb[2], f);
+                                shape->strokeFill(r, g, b, (*s)->style.strokeOpacity(frameNo));
+                            }
                         }
-                        cursor.x += (*s)->style.letterSpacing(frameNo);
+                        setColor = true;
```

Comment:
setColor represents the condition for both stroke and fill?

I understand this value is used to ensure interpolating from previous color, then probably boolean value should be devided:
- setFillColor
- setStrokeColor

Reply:
for now doesn't matter, since we don;t have flags for colors. as you noticed before, there is an issue with colors, we'll be solved in a separate pr.

Reply:
@mgrudzinska Please leave comments in the code (e.g., FIXME, TODO) for any known issues or tasks to do.